cclib/parser/logfileparser.py

```diff
@@ -476,12 +476,16 @@ def append_attribute(self, name: str, value: Any) -> None:
             self.set_attribute(name, [])
         getattr(self, name).append(value)
 
-    def extend_attribute(self, name: str, values: Iterable[Any]) -> None:
+    def extend_attribute(self, name: str, values: Iterable[Any], index: Optional[int] = None) -> None:
         """Appends an iterable of values to an attribute."""
         
         if not hasattr(self, name):
             self.set_attribute(name, [])
-        getattr(self, name).extend(values)
+
+        if isinstance(getattr(self, name), list) and index:
```

Comment:
```suggestion
        if isinstance(getattr(self, name), list) and index is not None:
```
Checks against `None` should always be explicit, unlike checking for an empty container or string.